src/System.Net.Security/tests/FunctionalTests/ServerRequireEncryptionTest.cs

```diff
@@ -82,8 +83,16 @@ public async Task ServerRequireEncryption_ClientNoEncryption_NoConnect()
                 await client.ConnectAsync(serverRequireEncryption.RemoteEndPoint.Address, serverRequireEncryption.RemoteEndPoint.Port);
                 using (var sslStream = new SslStream(client.GetStream(), false, AllowAnyServerCertificate, null, EncryptionPolicy.NoEncryption))
                 {
-                    await Assert.ThrowsAsync<IOException>(() =>
+                    if (PlatformDetection.SupportsTls13)
+                    {
+                        await Assert.ThrowsAsync<AuthenticationException>(() =>
                         sslStream.AuthenticateAsClientAsync("localhost", null, SslProtocolSupport.DefaultSslProtocols, false));
```

Comment:
Seems like maybe

```C#
Task authenticateAsClient =
    sslStream.AuthenticateAsClientAsync("localhost", null, SslProtocolSupport.DefaultSslProtocols, false));

if (PlatformDetection.SupportsTls13)
{
    await Assert.ThrowsAsync<AuthenticationException>(authenticateAsClient);
}
else
{
    await Assert.ThrowsAsync<IOException>(authenticateAsClient);
}
```

is more clearly doing the same work but expecting different exceptions based on the different environment state.